### PR TITLE
chore(docs): update for #18719

### PR DESCRIPTION
**Pull request by yoyo837**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
 #18719

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    update docs       |
| 🇨🇳 Chinese |  文档完善         |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/form/index.en-US.md](https://github.com/ant-design/ant-design/blob/18719-docs/components/form/index.en-US.md)
[View rendered components/form/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/18719-docs/components/form/index.zh-CN.md)

### CODE REVIEW

File: components/form/index.en-US.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ A form consists of one or more form fields whose type includes input, textarea,
 | layout | Define form layout | 'horizontal'\|'vertical'\|'inline' | 'horizontal' |  |
 | onSubmit | Defines a function will be called if form data validation is successful. | Function(e:Event) |  |  |
 | wrapperCol | (Added in 3.14.0. Previous version can only set on FormItem.) The layout for input controls, same as `labelCol` | [object](https://ant.design/components/grid/#Col) |  | 3.14.0 |
-| colon | change default props colon value of Form.Item | boolean | true | 3.15.0 |
+| colon | change default props colon value of Form.Item (only Form Layout: `Horizontal`) | boolean | true | 3.15.0 |
```
Comment (orzyyyy) [inline components/form/index.en-US.md]: only effective when prop `layout` is `horizontal`

File: components/form/index.zh-CN.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ title: Form
 | layout | 表单布局 | 'horizontal'\|'vertical'\|'inline' | 'horizontal' |  |
 | onSubmit | 数据验证成功后回调事件 | Function(e:Event) |  |  |
 | wrapperCol | （3.14.0 新增，之前的版本只能设置到 FormItem 上。）需要为输入控件设置布局样式时，使用该属性，用法同 labelCol | [object](https://ant.design/components/grid-cn/#Col) |  | 3.14.0 |
-| colon | 配置 Form.Item 的 colon 的默认值 | boolean | true | 3.15.0 |
+| colon | 配置 Form.Item 的 colon 的默认值 (仅在 Form Layout: `Horizontal` 有效) | boolean | true | 3.15.0 |
```
Comment (orzyyyy) [inline components/form/index.zh-CN.md]: 只有在 Form 属性 `layout` 为 `horizontal` 时有效

File: components/form/index.en-US.md (hunk 1 above)
Comment (yoyo837) [reply components/form/index.en-US.md]: 膜拜大佬 🥇

**Comment by afc163** [comment]
👍

**Pull request merged**